### PR TITLE
ENH Use File Title

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-linkfield/issues/240

### CODE REVIEW

File: tests/behat/features/create-edit-linkfield.feature [hunk 1]
```diff
@@ -147,7 +147,7 @@ I want to add links to pages, files, external URLs, email addresses and phone nu
 
     # Fourth link in multi link field
     And I should see "Link to a file" in the "[data-field-id='Form_EditForm_HasManyLinks'] .link-picker__link:nth-of-type(4)" element
-    And I should see "folder1/file1.jpg" in the "[data-field-id='Form_EditForm_HasManyLinks'] .link-picker__link:nth-of-type(4)" element
+    And I should see "File1" in the "[data-field-id='Form_EditForm_HasManyLinks'] .link-picker__link:nth-of-type(4)" element
```
Comment (ChloeSartorelli) [inline tests/behat/features/create-edit-linkfield.feature:150]: I think we should see both, right?
There's the default title which is now "File1", but we should also see the URL "folder1/file1.jpg" after "Link to a file"

image

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: tests/behat/features/create-edit-linkfield.feature (hunk 1 above)
Comment (emteknetnz) [reply tests/behat/features/create-edit-linkfield.feature:150]: I don't think that's correct, point of this PR is to not show the file url and to show the file title instead

This is how it looks with this PR with and without link text specified. Note that $file->Title is automatically generated in File::onBeforeWrite(), though in this case I've manually adjusted it

image

Comment (ChloeSartorelli) [reply tests/behat/features/create-edit-linkfield.feature:150]: Oh, interesting. I'm not sure that's the intended result? Might need to double check with Max or UX. My understanding is that only the default title was meant to change, not the text after "Link to a file".

File: src/Models/FileLink.php [hunk 2]
```diff
@@ -50,7 +50,7 @@ public function getDescription(): string
         if (!$file->canView()) {
             return _t(__CLASS__ . '.CANNOT_VIEW_FILE', 'Cannot view file');
         }
-        return $file->getFilename() ?? '';
+        return $file->Title ?? '';
```
Comment (ChloeSartorelli) [inline src/Models/FileLink.php]: I think this is meant to stay as is, and only `getDefaultTitle()` is meant to change.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: src/Models/FileLink.php (hunk 2 above)
Comment (emteknetnz) [reply src/Models/FileLink.php]: Ah yup makes sense, updated

File: tests/behat/features/create-edit-linkfield.feature (hunk 1 above)
Comment (emteknetnz) [reply tests/behat/features/create-edit-linkfield.feature:150]: Has been changed

**Review by GuySartorelli — APPROVED** [review]
LGTM

**Pull request merged**